src/jit/lowerxarch.cpp

```diff
@@ -1030,6 +1034,23 @@ void Lowering::TreeNodeInfoInitShiftRotate(GenTree* tree)
     GenTreePtr shiftBy = tree->gtOp.gtOp2;
     GenTreePtr source  = tree->gtOp.gtOp1;
 
+#ifdef _TARGET_X86_
+    // The first operand of a GT_LSH_HI and GT_RSH_LO oper is a GT_LONG so that
+    // we can have a three operand form. Increment the srcCount.
+    if (source->OperGet() == GT_LONG)
+    {
+        info->srcCount++;
+        source->gtLsraInfo.isDelayFree = true;
```

Comment:
I missed that you are marking source (which is GT_LONG)as isDelayFree, which is not required since it will be contained.

What is required here is

Mark one of sourceLo or sourceHi as isDelayFree depending on oper is LSH_HI or RSH_LO.
Mark source and tree as hasDelayFreeSrc=true

---
In reply to: [79076154](https://github.com/dotnet/coreclr/pull/7169#discussion_r79076154) [](ancestors = 79076154)